crates/exex/exex/src/wal/cache.rs

```diff
@@ -35,6 +35,11 @@ impl BlockCache {
         self.notification_max_blocks.is_empty()
     }
 
+    /// Returns the number of blocks in the cache.
+    pub(super) fn num_blocks(&self) -> u64 {
+        self.committed_blocks.len() as u64
```

Comment:
nit: we can remove a bunch of conversions by returning `usize` here

Reply:
yeah, i agree